keystrokes/src/components/Character/CharacterCreationModal.js

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react';
+import {
+	Modal,
+	Button,
+	TextField,
+	FormControl,
+	InputLabel,
+	Select,
+	MenuItem,
+	Grid,
+} from '@material-ui/core';
+import PropTypes from 'prop-types';
+import { characterDetails } from '../../config/constants';
+import styleGuide from '../../stylesheets/style';
+
+CharacterCreationModal.propTypes = {
+	handleSubmit: PropTypes.func.isRequired,
+	isOpen: PropTypes.bool.isRequired,
+	setIsOpen: PropTypes.bool.isRequired,
+};
+
+function canSave() {
+	return (
+		characterName === '' ||
+		characterSpec === '' ||
+		characterRace === '' ||
+		characterName === ''
+	);
+}
+
+function CharacterCreationModal({ handleSubmit, isOpen, setIsOpen }) {
+	const [characterName, setCharacterName] = useState('');
+	const [characterClass, setCharacterClass] = useState('');
+	const [characterRace, setCharacterRace] = useState('');
+	const [characterSpec, setCharacterSpec] = useState('');
+	const classes = styleGuide();
+
+	return (
+		<Modal
+			open={isOpen}
+			onClose={setIsOpen(true)}
```

Comment:
this looks fishy to me, `onClose={setIsOpen(true)}` wouldn't you want to set it to false on close?

Reply:
that probably explains why it isnt closing when i click outside the modal... good catch!